ee/control/consumers/notificationconsumer/notify_consumer.go

```diff
@@ -0,0 +1,167 @@
+package notificationconsumer
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	desktopRunner "github.com/kolide/launcher/ee/desktop/runner"
+	"github.com/kolide/launcher/pkg/osquery"
+	"go.etcd.io/bbolt"
+)
+
+// Consumes notifications from control server, tracks when notifications are sent to end user
+type NotificationConsumer struct {
+	db              *bbolt.DB
+	runner          notifier
+	logger          log.Logger
+	notificationTtl time.Duration // How long until a notification expires -- i.e. how long until we can re-send that identical notification
+}
+
+// The desktop runner fullfils this interface -- it exists primarily for testing purposes.
+type notifier interface {
+	SendNotification(title, body string) error
+}
+
+// Represents notification received from control server; SentAt is set by this consumer after sending.
+type notification struct {
+	Title  string    `json:"title"`
+	Body   string    `json:"body"`
+	UUID   string    `json:"uuid"`
```

Comment:
This is fine, but consider `id` -- we don't actually need formal uuids, and I'm not sure we want to encourage them. (uuid, ulid, random, whatever...)

Reply:
Sure, I'm fine with something instead of UUID -- why don't we want to encourage them, though? Do we just not want to be tied to expecting this string to be formatted a particular way, or is there another concern?

Reply:
No deep concerns. We just don't use uuids much, so it feels like a weird hint. (No reason for not using uuids, we mostly standardized on ulids, or other IDs)